lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module

```diff
@@ -453,6 +454,13 @@ function dosomething_campaign_build_campaigns_query($params = array()) {
     $query->condition('m.field_on_mobile_app_value', 1);
   }
 
+  if (isset($params['mobile_app_date'])) {
+    $query->join('field_data_field_mobile_app_date', 'm', 'm.entity_id = n.nid');
+    $query
+      ->condition('m.field_mobile_app_date_value', dosomething_helpers_convert_date($params['mobile_app_date']), '<=')
+      ->condition('m.field_mobile_app_date_value2', dosomething_helpers_convert_date($params['mobile_app_date']), '>=');
```

Comment:
I might just be confused, but wouldn't this just grab everything? Less than or equal to the specified date, and greater than or equal to the specified date?